table/arrow_utils.go

```diff
@@ -813,7 +823,26 @@ func (a *arrowProjectionVisitor) Struct(st iceberg.StructType, structArr arrow.A
 		}
 	}
 
-	return retOrPanic(array.NewStructArrayWithFields(fieldArrs, fields))
+	var nullBitmap *memory.Buffer
+	if structArr.NullN() > 0 {
```

Comment:
Would not be cleaner to reverse condition and return 
```suggestion
	if structArr.NullN() <= 0 {
	    return retOrPanic(array.NewStructArrayWithFieldsAndNulls(fieldArrs, fields,
		  nullBitmap, structArr.NullN(), 0))
```

Reply:
I don't think that would make it any cleaner really. Unless I'm missing something, all that would do is force us to duplicate the return line and only remove one level of braces. 

Am I misunderstanding?

Reply:
You're right — I analyzed this logic incorrectly.